src/pseudopeople/noise_functions.py

```diff
@@ -550,35 +552,48 @@ def make_typos(
     include_token_probability_level = 0.1
     rng = np.random.default_rng(seed=get_hash(f"{randomness_stream.seed}_make_typos"))
 
-    # Make all strings the same length by padding with spaces
-    max_str_length = column.str.len().max()
-    same_len_col = column.str.pad(max_str_length, side="right")
+    # https://stackoverflow.com/a/9493192/
+    # NOTE: Surprisingly, casting this to "object" (Python string type) seems to make
+    # the rest of this method faster. Without this cast, you have to use np.char.add to concatenate,
+    # which seems very slow for some reason?
+    same_len_col_exploded = (
+        column.values.astype(str).view("U1").reshape((len(column), -1)).astype("object")
```

Comment:
Is there a reason you have both `astype(str)` and `astype("object")` in this line? Aren't they the same thing?

Reply:
No -- `astype(str)` turns it into a NumPy fixed-length string type, not a Python string type. This is counterintuitive so I'll add a comment.

Reply:
Looks like if you try astype back to `str` it remains dtype '<U1'. The unicode types are weird

Reply:
I think that's because it only has one-character strings in it. `.astype(str)` turns it into a Unicode string type like `<U#` (the `<` means little-endian, `#` would be replaced the length of the longest string). For example `np.array(["foo", "bar"]).astype("object").astype(str)` turns it back to dtype `<U3` because "foo" and "bar" each have three characters.